### ISSUE TITLE
angular cli with projects assets : root dir not correctly used

### ISSUE THREAD

**Issue by headonkeyboard** [storybooks/storybook#3864]
### Bug or support request summary

Expecting loading scripts from node_modules folder by adding path to "scripts" in angular.json file. File is not found during storybook build and serve but correctly imported on angular app.

### Steps to reproduce

1. Create a new project with angular cli
2. Add another application to workspace : `ng generate application my-other-app`
3. Install a third party library like jquery `npm install jquery`
4. Add jquery script to project scripts
```
{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "my-other-app": {
      "root": "projects/my-other-app/",
      "sourceRoot": "projects/my-other-app/src",
      "projectType": "application",
      ...
      },
      "architect": {
        "build": {
          ...
          "options": {
            ...
            "scripts": ["node_modules/jquery/dist/jquery.min.js"]
          },
         ...
  },
  "defaultProject": "my-other-app"
}
```

When running, storybook expects file to exist under `projects/my-other-app/node_modules/jquery/dist/jquery.min.js` instead of `node_modules/jquery/dist/jquery.min.js`. This behavior doesn't happen when running angular app.

### Please specify which version of Storybook and optionally any affected addons that you're running

- @storybook/angular: "4.0.0-alpha.12"

**Comment by marshallmick007** [comment]
This happens with "styles" as well.

My angular.json
```
{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "",
  "defaultProject": "dashboard-app",
  "projects": {
    "dashboard-app": {
      "root": "apps/dashboard-app/",
      "sourceRoot": "apps/dashboard-app/src",
      "projectType": "application",
      "prefix": "dashboard-app",
      "schematics": {
        "@schematics/angular:component": {
          "styleext": "scss"
        }
      },
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            ...
            "styles": [
              "apps/dashboard-app/src/styles.scss"
            ],
            "scripts": []
          }
	...
}
```

This config builds with `ng build` just fine, but when running storybook, with `yarn run storybook`

```
ERROR in multi ./apps/dashboard-app/apps/dashboard-app/src/styles.scss
Module not found: Error: Can't resolve 'X:\work\angular\dashboard-app-workspace\apps\dashboard-app\apps\dashboard-app\src\styles.scss' in 'X:\work\angular\dashboard-app-workspace'
 @ multi ./apps/dashboard-app/apps/dashboard-app/src/styles.scss
```

The storybook webpack settings do not look like they take angular-cli's `root` path into consideration.

**Comment by marshallmick007** [comment]
I tossed some crude logging into the `@angular/devkit` code and see what is likely the culprit

Running an `ng-build`, the config object passed into `getStylesConfig()` looks like this

 ## ng build

 ```
  { root: 'C:\\work\\angular\\dashboard-app-workspace',
  projectRoot: 'C:\\work\\angular\\dashboard-app-workspace\\apps\\dashboard-app',
  buildOptions:
   { outputPath: 'dist/apps/dashboard-app',
     index: 'apps/dashboard-app/src/index.html',
     main: 'apps/dashboard-app/src/main.ts',
     polyfills: 'apps/dashboard-app/src/polyfills.ts',
     tsConfig: 'apps/dashboard-app/tsconfig.app.json',
     assets: [ [Object], [Object] ],
     styles: [ 'apps/dashboard-app/src/styles.scss' ],
}
```	 
but when running a `storybook` build, the `root` and `projectRoot` are different, and I am assuming this is causing `@angular/devkit` to spit out an inaccurate path for styles (and likely scripts as well)

## storybook

```
{ root: 'apps/dashboard-app/',
  projectRoot: 'C:\\work\\angular\\dashboard-app-workspace',
  supportES2015: false,
  tsConfig: { options: {}, fileNames: [], errors: [] },
  tsConfigPath: 'C:\\work\\angular\\dashboard-app-workspace\\src\\tsconfig.app.json',
  buildOptions:
   { outputPath: 'dist/apps/dashboard-app',
     index: 'apps/dashboard-app/src/index.html',
     main: 'apps/dashboard-app/src/main.ts',
     polyfills: 'apps/dashboard-app/src/polyfills.ts',
     tsConfig: 'apps/dashboard-app/tsconfig.app.json',
     assets: [ [Object], [Object] ],
     styles: [ 'apps/dashboard-app/src/styles.scss' ],
     scripts: [] } }
```

**Comment by igor-dv** [comment]
Would you lile to PR this ?

**Comment by marshallmick007** [comment]
@igor-dv Perhaps with a little direction. I am a bit concerned about backwards compatibility with a change such as this, and the potential for breaking Angular 5 or previous versions. Does storybook 4.0 desire to remain backward compatible or is its goal to support Angular 6 and greater in the future?

**Comment by igor-dv** [comment]
Storybook v4 is a breaking release, it also supports webpack v4, which means it integrates with other tools that also should support webpack v4. That's why we adopted angular-cli v6... So, I assume it can't be backward compatible easily 🤷‍♂️

**Comment by matthewharwood** [comment]
>_< @marshallmick007 Tried replacing that line and i get hun on this unnamed compact pluginTypeError : missing Deps.sume is not a function

```
matth@DESKTOP-PUQ7U15 MINGW64 ~/WebstormProjects/morningharwood (feature/storybooks)
$ npm run storybook

> morningharwood@0.0.0 storybook C:\Users\matth\WebstormProjects\morningharwood
> start-storybook -p 9001 -c .storybook

info => Loading angular-cli config.
info @storybook/angular v4.0.0-alpha.15
info
info => Get angular-cli webpack config.
info => Using default webpack setup based on "angular-cli".
 10% building modules 3/3 modules 0 activeℹ ｢wdm｣: wait until bundle finished:
(node:20264) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead              95% emitting unnamed compat pluginTypeError: missingDeps.some is not a function
    at compiler.plugin (C:\Users\matth\WebstormProjects\morningharwood\node_modules\react-dev-utils\WatchMissingNodeModulesPlugin.js:25:23)
    at _err2 (eval at create (C:\Users\matth\WebstormProjects\morningharwood\node_modules\tapable\lib\HookCodeFactory.js:24:12), <anonymous>:33:1)
    at Promise.resolve.then.then.then.then.then.then.then.then.catch.then.then.then (C:\Users\matth\WebstormProjects\morningharwood\node_modules\html-webpack-plugin\index.js:224:11)
    at <anonymous>
```
I'm on windows >_< any ideas?

**Comment by Hypnosphi** [comment]
Released as [`4.0.0-alpha.20`](https://github.com/storybooks/storybook/releases/tag/v4.0.0-alpha.20)
